### PR TITLE
Separate init blob upload

### PR DESCRIPTION
**Pull request by jlhawn**
Pushing a v2 image layer has two steps:
- POST to get a new upload URL
- PUT to that upload URL

We were previously not checking the response code of
the POST request and the PUT would fail in weird ways.

### CODE REVIEW

**Comment by dmp42** [comment]
@jlhawn build fail

**Comment by dmcgowan** [comment]
LGTM

**Comment by stevvooe** [comment]
LGTM

**Comment by dmp42** [comment]
LGTM

**Comment by tiborvass** [comment]
I confirm that this resolved an issue i had pushing.

File: registry/session_v2.go [hunk 1]
```diff
@@ -262,6 +248,51 @@ func (r *Session) PutV2ImageBlob(ep *Endpoint, imageName, sumType, sumStr string
...
+
+		log.Debugf("Unexpected response from server: %q %#v", errBody, res.Header)
+		return "", utils.NewHTTPRequestError(fmt.Sprintf("Server error: unexpected %d response status trying to initiate upload of %s", res.StatusCode, imageName), res)
+	}
+
+	if location = res.Header.Get("Location"); location == "" {
+		return "", fmt.Errorf("registry did not return a Location header for resumable blob upload for image %s", imageName)
+	}
+
+	return
```
Comment (jfrazelle) [inline registry/session_v2.go]: I'm not really a fan of empty return its hard to follow, but I know this is a nit
